addons/website_sale/tests/test_website_sale_pricelist.py

```diff
@@ -522,3 +528,22 @@ def test_property_product_pricelist_multi_company(self):
         # also read a pricelist from another company if that company is the one
         # from the currently visited website.
         self.env(user=self.env.ref('base.user_demo'))['product.pricelist'].browse(demo_pl.id).name
+
+    def test_archive_pricelist_1(self):
+        ''' Test that when a pricelist is archived, the check that verify that
+            all website have at least one pricelist have access to all
+            pricelists (without considering the current allowed companies).
+        '''
+        # For testing purpose, remove unused websites if available.
+        self.env['website'].search([('id', 'not in', [self.website.id, self.website2.id])]).unlink()
```

Comment:
You might need to handle the PL before unlinking the website, runbot is red

Reply:
@rdeodoo  Indeed. I had to create a temporary 'public' pricelist to be able to unlink other websites. Updated :+1: